File: zig/raindrops/raindrops.zig
const fmt = @import("std").fmt;

pub fn convert(buffer: []u8, n: u32) []const u8 {
    const res = fmt.bufPrint(buffer, "{s}{s}{s}", .{
        if (n % 3 == 0) "Pling" else "",
        if (n % 5 == 0) "Plang" else "",
        if (n % 7 == 0) "Plong" else "",
    }) catch unreachable;
    return if (res.len > 0) res else fmt.bufPrint(buffer, "{d}", .{n}) catch unreachable;
}

File: euphoria/raindrops/raindrops.ex
include std/math.e
include std/utils.e
include std/text.e

public function raindrops(atom n)
  sequence res = iif(mod(n, 3) = 0, "Pling", "") & iif(mod(n, 5) = 0, "Plang", "") & iif(mod(n, 7) = 0, "Plong", "")
  return iif(equal(res, ""), sprint(n), res)
end function
